rust/core-lib/src/find.rs

```diff
@@ -226,31 +222,29 @@ impl Find {
     pub fn next_occurrence(&self, text: &Rope, reverse: bool, wrapped: bool, sel: (usize, usize)) -> Option<SelRegion> {
         let (sel_start, sel_end) = sel;
 
-        self.occurrences.as_ref().and_then(|occurrences| {
-            if occurrences.len() == 0 {
-                return None;
-            }
-
-            if reverse {
-                let next_occurrence = match sel_start.checked_sub(1) {
-                    Some(search_end) => occurrences.regions_in_range(0, search_end).last(),
-                    None => None
-                };
+        if self.occurrences.len() == 0 {
+            return None;
+        }
 
-                if next_occurrence.is_none() && !wrapped {
-                    return occurrences.regions_in_range(0, text.len()).last();
-                }
+        if reverse {
+            let next_occurrence = match sel_start.checked_sub(1) {
+                Some(search_end) => self.occurrences.regions_in_range(0, search_end).last(),
+                None => None
+            };
 
-                next_occurrence
-            } else {
-                let next_occurrence = occurrences.regions_in_range(sel_end + 1, text.len()).first();
+            if next_occurrence.is_none() && !wrapped {
+                return self.occurrences.regions_in_range(0, text.len()).last().cloned();
+            }
 
-                if next_occurrence.is_none() && !wrapped {
-                    return occurrences.regions_in_range(0, text.len()).first();
-                }
+            next_occurrence.cloned()
+        } else {
+            let next_occurrence = self.occurrences.regions_in_range(sel_end + 1, text.len()).first();
 
-                next_occurrence
+            if next_occurrence.is_none() && !wrapped {
+                return self.occurrences.regions_in_range(0, text.len()).first().cloned();
             }
-        }).cloned()
+
+            next_occurrence.cloned()
+        }
```

Comment:
I wonder if there is a more elegant way than adding `cloned()` to each occurrence

Reply:
I'd be fine with `SelRegion` being Copy, if that works out.

Reply:
`Copy` is already derived for `SelRegion`. The problem was that instead of `Option<SelRegion>` the derived clone method will create a reference `Option<&SelRegion>` (I think it has something to do with this: https://github.com/rust-lang/rust/issues/26925). I could implement `Copy` manually, but not sure if that is better.